sdk/include/opentelemetry/sdk/logs/attribute_utils.h

```diff
@@ -0,0 +1,118 @@
+
+/**
+ * Creates an owned copy (LogRecordAttributeValue) of a non-owning AttributeValue.
+ */
+struct AttributeConverter
+{
+  LogRecordAttributeValue operator()(bool v) { return LogRecordAttributeValue(v); }
+  LogRecordAttributeValue operator()(int32_t v) { return LogRecordAttributeValue(v); }
+  LogRecordAttributeValue operator()(uint32_t v) { return LogRecordAttributeValue(v); }
+  /*LogRecordAttributeValue operator()(int v)
+  {
+    return LogRecordAttributeValue(static_cast<int64_t>(v));
+  }*/
+  LogRecordAttributeValue operator()(int64_t v) { return LogRecordAttributeValue(v); }
+  LogRecordAttributeValue operator()(uint64_t v) { return LogRecordAttributeValue(v); }
+  LogRecordAttributeValue operator()(double v) { return LogRecordAttributeValue(v); }
+  LogRecordAttributeValue operator()(nostd::string_view v)
+  {
+    return LogRecordAttributeValue(std::string(v));
+  }
+  LogRecordAttributeValue operator()(nostd::span<const bool> v) { return convertSpan<bool>(v); }
+  LogRecordAttributeValue operator()(nostd::span<const int32_t> v)
+  {
+    return convertSpan<int32_t>(v);
+  }
+  LogRecordAttributeValue operator()(nostd::span<const uint32_t> v)
+  {
+    return convertSpan<uint32_t>(v);
+  }
+  LogRecordAttributeValue operator()(nostd::span<const int64_t> v)
+  {
+    return convertSpan<int64_t>(v);
+  }
+  LogRecordAttributeValue operator()(nostd::span<const uint64_t> v)
+  {
+    return convertSpan<uint64_t>(v);
+  }
+  LogRecordAttributeValue operator()(nostd::span<const double> v) { return convertSpan<double>(v); }
+  LogRecordAttributeValue operator()(nostd::span<const nostd::string_view> v)
+  {
+    return convertSpan<std::string>(v);
+  }
+
+  template <typename T, typename U = T>
+  LogRecordAttributeValue convertSpan(nostd::span<const U> vals)
+  {
+    const std::vector<T> copy(vals.begin(), vals.end());
+    return LogRecordAttributeValue(std::move(copy));
+  }
+};
+
+/**
+ * Class for storing attributes.
+ */
+class AttributeMap
+{
+public:
+  // Contruct empty attribute map
+  AttributeMap(){};
+
+  // Contruct attribute map and populate with attributes
+  AttributeMap(const opentelemetry::common::KeyValueIterable &attributes)
+  {
+    attributes.ForEachKeyValue([&](nostd::string_view key,
+                                   opentelemetry::common::AttributeValue value) noexcept {
+      SetAttribute(key, value);
+      return true;
+    });
+  }
+
+  const std::unordered_map<std::string, LogRecordAttributeValue> &GetAttributes() const noexcept
+  {
+    return attributes_;
+  }
+
+  void SetAttribute(nostd::string_view key,
+                    const opentelemetry::common::AttributeValue &value) noexcept
+  {
+    attributes_[std::string(key)] = nostd::visit(converter_, value);
+  }
+
+private:
+  std::unordered_map<std::string, LogRecordAttributeValue> attributes_;
+  AttributeConverter converter_;
+};
+}  // namespace logs
+}  // namespace sdk
+OPENTELEMETRY_END_NAMESPACE
```

Comment:
Is it possible we can share this class across trace and logs, and move it to `sdk/common`. Probably not as part of this PR if it's huge task, we can have a issue open for it.

Reply:
I can fix this in a future PR! (If I don't have time I will create an issue for it)

Reply:
@lalitb @jsuereth Just an update on this-  I've moved the `attribute_utils.h` file from `logs` to `common` for this PR, and renamed `LogRecordAttributeValue` to `OwnedAttributeValue`. All the necessary naming changes for `LogRecordAttributeValue` have been done in this PR. 

Seeing as this PR seems to already be running on the long side, and also unrelated to traces/spans, I've filed the traces area needing to be refactored as an issue here: #442 , which I may take a look at when working on the OStream exporter for logs, or if not would probably be a good first issue for someone to tackle as well